lib/al/Library/Matrix/MatrixUtil.h

```diff
@@ -0,0 +1,88 @@
+void makeMtxQuatPos(sead::Matrix34f*, sead::Quatf const&, sead::Vector3f const&);
+void makeMtxQuatScalePos(sead::Matrix34f*, sead::Quatf const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxQuatScalePos(sead::Matrix44f*, sead::Quatf const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxFrontUpPos(sead::Matrix34f*, sead::Vector3f const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxFrontSidePos(sead::Matrix34f*, sead::Vector3f const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxUpFrontPos(sead::Matrix34f*, sead::Vector3f const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxUpSidePos(sead::Matrix34f*, sead::Vector3f const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxSideUpPos(sead::Matrix34f*, sead::Vector3f const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxSideFrontPos(sead::Matrix34f*, sead::Vector3f const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxFollowTarget(sead::Matrix34f*, sead::Matrix34f const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxProj(sead::Matrix44f*, sead::Vector2<f32> const&, sead::Vector3f const&, sead::Vector3f const&);
+void makeMtxProjFromQuatPoseUp(sead::Matrix44f*, sead::Quatf const&, sead::Vector2<f32> const&, sead::Vector3f const&);
+void makeMtxProjFromQuatPoseFront(sead::Matrix44f*, sead::Quatf const&, sead::Vector2<f32> const&, sead::Vector3f const&);
+void makeMtxProjFromQuatPoseSide(sead::Matrix44f*, sead::Quatf const&, sead::Vector2<f32> const&, sead::Vector3f const&);
+void makeMtxProjFromUp(sead::Matrix44f*, sead::Vector2<f32> const&, sead::Vector3f const&);
+void rotateMtxXDirDegree(sead::Matrix34f*, sead::Matrix34f const&, f32);
+void rotateMtxYDirDegree(sead::Matrix34f*, sead::Matrix34f const&, f32);
+void rotateMtxZDirDegree(sead::Matrix34f*, sead::Matrix34f const&, f32);
+void rotateMtxCenterPosXDirDegree(sead::Matrix34f*, sead::Matrix34f const&, sead::Vector3f const&, f32);
+void rotateMtxCenterPosAxisDegree(sead::Matrix34f*, sead::Matrix34f const&, sead::Vector3f const&, sead::Vector3f const&, f32);
+void rotateMtxCenterPosYDirDegree(sead::Matrix34f*, sead::Matrix34f const&, sead::Vector3f const&, f32);
+void rotateMtxCenterPosZDirDegree(sead::Matrix34f*, sead::Matrix34f const&, sead::Vector3f const&, f32);
+void rotateMtxCenterPosQuat(sead::Matrix34f*, sead::Matrix34f const&, sead::Vector3f const&, sead::Quatf const&);
+void turnMtxXDirDegree(sead::Matrix34f*, sead::Matrix34f const&, sead::Vector3f const&, f32);
+void turnMtxYDirDegree(sead::Matrix34f*, sead::Matrix34f const&, sead::Vector3f const&, f32);
+void turnMtxZDirDegree(sead::Matrix34f*, sead::Matrix34f const&, sead::Vector3f const&, f32);
+void calcCameraPosFromViewMtx(sead::Vector3f*, sead::Matrix34f const&);
+void calcMtxMul(sead::Vector3f*, sead::Matrix34f const&, sead::Vector3f const&);
+// void calcMtxMul(sead::Vector3f *,al::Matrix43<f32> const&,sead::Vector3f const&);
```

Comment:
Why is this (and others) commented out?

Reply:
al::Matrix43 is a templated class, I think it's best that this is created properly when somebody knows how it works. As for the ones using the `nn::util::neon` namespace, I've forward declared those now.